src/webgpu/util/compare.ts

```diff
@@ -73,18 +73,20 @@ export function compare(got: Value, expected: Value, cmpFloats: FloatMatch): Com
     const gTy = got.type;
     const eTy = expected.type;
     if (gTy !== eTy) {
-      return {
-        matched: false,
-        got: `${Colors.red(gTy.toString())}(${got})`,
-        expected: `${Colors.red(eTy.toString())}(${expected})`,
-      };
+      if (!isFloatValue(got) || !isFloatValue(expected)) {
```

Comment:
Not sure I understand this, previously if the numbers were not equal we'd fail, now we only fail if one of the things isn't a float?

Reply:
I am loosening the restriction so that different float types will pass. I will take another look at if I can express this more clearly, since I agree it is a little obtuse

Reply:
Pulling inner check into a variable, that allows for a more readable single check